pinot-common/src/main/java/org/apache/pinot/common/utils/ServiceStatus.java

```diff
@@ -443,8 +467,9 @@ private StatusDescriptionPair evaluateResourceStatus(String resourceName) {
       // external view or went to ERROR state (which means that we tried to load the segments/resources but failed for
       // some reason)
       Map<String, String> partitionStateMap = getPartitionStateMap(helixState);
-      for (String partitionName : idealState.getPartitionSet()) {
-        String idealStateStatus = idealState.getInstanceStateMap(partitionName).get(_instanceName);
+      for (String partitionName : partitionSet) {
+        String idealStateStatus = partitionToIdealStateStatusMap.getOrDefault(partitionName,
+            idealState.getInstanceStateMap(partitionName).get(_instanceName));
```

Comment:
(nit) We can just use get() instead of getOrDefault() since the map value has been set as `idealState.getInstanceStateMap(partitionName).get(_instanceName)` on L448?

Reply:
I think it's needed, otherwise we'll skip checking some partition's statuses as we break when collecting ideal state statuses on L453. Added a test case to illustrate this

Reply:
Got it, make sense.